### PR TITLE
[stable/envoy]: optional feature to generate envoy.yaml from golang template

### PR DESCRIPTION
**Pull request by mumoshu**
This feature enables you to make your values.yaml a bit more DRY by referencing chart values from within the envoy.yaml template by using the `{{ .Values.foo.bar }}`.

The following example shows how you can reuse the port numbers defined in `values.yaml`:

- `ports.admin.containerPort`
- `ports.n0.contianerPort`

from within the envoy.yaml template by writing the below template expressions in `templates.envoy\.yaml`:

- `{{ .Values.ports.admin.containerPort }}`
- `{{ .Values.ports.n0.containerPort }}`

```
ports:
  admin:
    containerPort: 9901
    protocol: TCP
  n0:
    containerPort: 10000
    protocol: TCP

# snip

templates:
  envoy.yaml: |-
    ## refs:
    ## - https://www.envoyproxy.io/docs/envoy/latest/start/start#quick-start-to-run-simple-example
    ## - https://raw.githubusercontent.com/envoyproxy/envoy/master/configs/google_com_proxy.v2.yaml
    admin:
      access_log_path: /dev/stdout
      address:
        socket_address:
          address: 0.0.0.0
          port_value: {{ .Values.ports.admin.containerPort }}

    static_resources:
      listeners:
      - name: listener_0
        address:
          socket_address:
            address: 0.0.0.0
            port_value: {{ .Values.ports.n0.containerPort }}
        filter_chains:
        - filters:
          - name: envoy.http_connection_manager
            config:
              access_log:
              - name: envoy.file_access_log
                config:
                  path: /dev/stdout
              stat_prefix: ingress_http
              route_config:
                name: local_route
                virtual_hosts:
                - name: local_service
                  domains: ["*"]
                  routes:
                  - match:
                      prefix: "/"
                    route:
                      host_rewrite: www.google.com
                      cluster: service_google
              http_filters:
              - name: envoy.router
      clusters:
      - name: service_google
        connect_timeout: 0.25s
        type: LOGICAL_DNS
        dns_lookup_family: V4_ONLY
        lb_policy: ROUND_ROBIN
        hosts:
          - socket_address:
              address: google.com
              port_value: 443
        tls_context:
          sni: www.google.com
```

Signed-off-by: Yusuke KUOKA <ykuoka@gmail.com>

### CODE REVIEW

**Comment by mumoshu** [comment]
@josdotso Hey! Thanks for maintaining this awesome chart.

This optional feature reduces repetitions in your values.yaml. Being optional, the default values.yaml doesn't include it, meaning it is completely backward-compatible.

Would you mind reviewing this?

File: stable/envoy/README.md [hunk 1]
```diff
@@ -2,4 +2,33 @@
...
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+## Configuration
+
+The following table lists the configurable parameters of the envoy chart and their default values.
+
+Parameter | Description | Default
+--- | --- | ---
+`files.envoy\.yaml` | content of a full envoy configuration file as documented in https://www.envoyproxy.io/docs/envoy/latest/configuration/configuration | See [values.yaml](values.yaml)
+`templates.envoy\.yaml` | golang template of a full configuration file. Use the `{{ .Values.foo.ba }}` syntax to embed chart values | See [values.yaml](values.yaml)
```
Comment (ghost) [inline stable/envoy/README.md]: Did you mean `.Values.foo.bar` here?

**Review by ghost — APPROVED** [review]
LGTM. Glad to accept another PR to add you as a maintainer of this chart.

Thanks!
-Joshua

File: stable/envoy/README.md (hunk 1 above)
Comment (mumoshu) [reply stable/envoy/README.md]: Good catch! Yes, I fixed it in the latest commit

**Comment by mumoshu** [comment]
@josdotso Thanks for reviewing ☺️ 

I think this is good to go now. As you're the owner of this chart, I believe you can say `/lgtm` followed by `/ok-to-test` to merge this.

**Pull request merged**